File: assignment1/p2.py
# To calculate the factorial of a number n and to give an error when the input is a negative number.
n = int(input("Enter a number.\n"))
if n<0:
    print("Factorial for negative numbers is undefined!!!\n")
    exit()
elif n==0:
    prod = 1
    print("The factorial of", n, "is", prod)
else:
    prod = 1
    for i in range(1,n+1):
        prod *= i
    print("The factorial of", n, "is", prod)

File: assignment1/p2.c
// To find the factorial of a number n and to trap cases where the number is negative
#include <stdio.h>
#include <stdlib.h>

void main()
{
    int n, prod;
    prod = 1;
    printf("Enter a number\n");
    scanf("%d", &n);
    if(n<0)
    {
        printf("This number is NEGATIVE. Please enter a another number!!!");
        exit(0);
    }
    else if(n==0)
    {
        printf("FACTORIAL: 1");
    }
    else
    {
        for(int i=1; i<=n; i++)
        {
            prod *= i;
        }
    }
    printf("The factorial is %d", prod);
}
